### PR TITLE
Remove `use_gzip` from HTMLEditorField since it's been removed by TinyMCE codebase

### PR DESCRIPTION
**Pull request by phalkunz**
Original part of https://github.com/silverstripe/silverstripe-admin/issues/158

FYI: TinyMCE has removed it https://github.com/tinymce/tinymce-docs/pull/460

### CODE REVIEW

**Comment by tractorcow** [comment]
I'm not sure we can remove this in beta. Removal of API / features is now 5.0 at the earliest. In which case, we should remove that entire class (as it's redundant).

**Comment by flamerohr** [comment]
Cleaned up rebase to master :)

**Comment by dhensby** [comment]
> I'm not sure we can remove this in beta. Removal of API / features is now 5.0 at the earliest. In which case, we should remove that entire class (as it's redundant).

if this is a defunct API which has no impact what-so-ever then I feel we should remove it despite being in beta.

The beta should give us some flexibility around this and keeping a useless API around for 3 years feels like dogma over reason.

**Comment by tractorcow** [comment]
@dhensby if you get another core committer to second that then we can remove it in 4. I will admit to being a bit dogmatic about semver. :)

**Comment by dhensby** [comment]
@tractorcow I never got round to reading this notification until now and I guess now we're in RC phase it's going to be even harder to get this removed from 4.x ;)

Are `private statics` an API at all? I still think this can be removed in the RC...

@chillu - what do you think to this?

**Comment by tractorcow** [comment]
> Are private statics an API at all? I still think this can be removed in the RC...

Yes it's an API. I'd feel better about removing this on master branch.

**Comment by sminnee** [comment]
If this is broken in SS4 then I think removing it is a bugfix, not an API change. I'd :+1: to removing it and the class. The config option can still be set and perform a no-op if the private static is removed so that's fine.

If we really want to avoid deleting the class, then I would just clearing out the classes content to only throw a deprecation warning saying "this doesn't work" and deferring to default behaviour.

**Comment by tractorcow** [comment]
Is there an actual verified report that it's broken? I understood this as a removal of an undesirable feature.

**Comment by sminnee** [comment]
Oh, right, I just assumed. If it's still working then I would bump to master too.

**Comment by dhensby** [comment]
The feature has been removed by tinymce, so whilst our class does *something* it's not usable for its intended purpose.

**Comment by tractorcow** [comment]
> The feature has been removed by tinymce, so whilst our class does something it's not usable for its intended purpose.

Are you talking about tiny_mce_gzip.php? It's still in thirdparty dir, and it is usable for it's intended purpose. Can you please elaborate on "not usable"?

**Comment by dhensby** [comment]
See https://github.com/tinymce/tinymce-docs/pull/460

**Comment by tractorcow** [comment]
I've seen that; That's not retroactively removing / rendering inoperable the php file from every current tinymce installation out there, including the one bundled with our last beta.

**Comment by maxime-rainville** [comment]
Related to https://github.com/silverstripe/silverstripe-admin/issues/832

File: src/Forms/HTMLEditor/HTMLEditorField.php [hunk 1]
```diff
@@ -28,14 +28,6 @@ class HTMLEditorField extends TextareaField
 
     protected $schemaComponent = 'HtmlEditorField';
 
-    /**
-     * Use TinyMCE's GZIP compressor
-     *
-     * @config
-     * @var bool
-     */
-    private static $use_gzip = true;
```
Comment (maxime-rainville) [inline src/Forms/HTMLEditor/HTMLEditorField.php]: I think removing this line is OK as there's no way to access it directly and setting a config flag for a value that doesn't exist doesn't cause an error.

**Comment by chillu** [comment]
Tested locally, added changelog entry, and merged.

**Pull request merged**